rutil/ssl/SHA1Stream.hxx

```diff
@@ -51,13 +41,8 @@ class SHA1Buffer : public std::streambuf
       virtual int sync();
       virtual int overflow(int c = -1);
    private:
-      // !kh!
-      // used pointers to keep the same object layout.
-      // this adds overhead, two additional new/delete.
-      // could get rid of the overhead if, sizeof(SHA_CTX) and SHA_DIGEST_LENGTH are known and FIXED.
-      // could use pimpl to get rid of one new/delete pair.
-      std::unique_ptr<SHA_CTX> mContext;
-      std::vector<char> mBuf;
+      EVP_MD_CTX* mContext;
+      char mBuf[64];
```

Comment:
Could you include ` #include <openssl/evp.h>` and use:
```suggestion
      char mBuf[EVP_MAX_MD_SIZE];
```

Reply:
This buffer is not used to store the digest, it is used for IO. E.g. if you output an integer into the stream, this is where it gets formatted.

The buffers for the digest are allocated in `SHA1Buffer::getHex` and `SHA1Buffer::getBin`, on the stack.